samples/unit-tests/series-pie/innersize/demo.js

```diff
@@ -86,3 +86,76 @@ QUnit.test(
         );
     }
 );
+
+QUnit.test('Thickness cannot be greater than size.(#6647)', assert => {
+    let clicks = 0;
+    const chart = new Highcharts.chart('container', {
+        chart: {
+            type: 'pie'
+        },
+        plotOptions: {
+            pie: {
+                thickness: 100,
+                events: {
+                    click: () => {
+                        clicks++;
+                    }
+                }
+            }
+        },
+        series: [
+            {
+                data: [
+                    ['Firefox', 44.2],
+                    ['IE7', 26.6],
+                    ['IE6', 20],
+                    ['Chrome', 3.1],
+                    ['Other', 5.4]
+                ]
+            }
+        ]
+    });
+
+    const series = chart.series[0],
+        thickness = series.options.thickness,
+        size = series.center[2];
+
+    assert.strictEqual(
+        thickness,
+        series.points[0].shapeArgs.r - series.points[0].shapeArgs.innerR,
+        'Thickness should be same as points distance.'
+    );
+
+    assert.strictEqual(
+        thickness + series.points[0].shapeArgs.innerR,
+        size / 2,
+        'Thickness should not be greater than size.'
+    );
+
+    const controller = new TestController(chart);
+
+    controller.moveTo(
+        chart.plotLeft + chart.series[0].center[0] - 55,
+        chart.plotTop + chart.series[0].center[1] + 5
+    );
+
+    controller.click(
+        chart.plotLeft + chart.series[0].center[0] - 55,
+        chart.plotTop + chart.series[0].center[1] + 5
+    );
+    assert.strictEqual(
+        clicks, 1, 'Clicking on a side of a pie slice should fire click event.'
+    );
+
+    series.update({
+        thickness: undefined
+    });
+    assert.ok(true, `After disable thickness, shouldn't be displayed.`);
```

Comment:
`assert.ok(true);`  isn't testing anything specific (it fails only when errors occur). When updating **any options**, the actual expected result should be tested (you can reuse the above asserts).